ow_plexil/src/plexil-adapter/FaultDependencies.cpp

```diff
@@ -247,10 +247,12 @@ void FaultDependencies::parseXML(const std::string &file){
   }
   else
   {
-      ROS_INFO_STREAM("XML [" << file_name << "] parsed with errors, attr value: [" << doc.child("node").attribute("attr").value() << "]\n");
-      ROS_INFO_STREAM("Error description: " << result.description() << "\n");
-      ROS_INFO_STREAM("Error offset: " << result.offset << " (error at [..." << (file_name + result.offset) << "]\n\n");
-      return;
+      ROS_WARN_STREAM("XML [" << file_name << "] parsed with errors, attr value: [" << doc.child("node").attribute("attr").value() << "]\n");
+      ROS_WARN_STREAM("Error description: " << result.description() << "\n");
+      ROS_WARN_STREAM("Error offset: " << result.offset << " (error at [..." << (file_name + result.offset) << "]\n\n");
+      // If we are unable to parse or the file doesnt exist, abort and cause a crash.
+      // This was requested so that the user must restart with a valid XML file and cannot ignore.
+      std::abort();
```

Comment:
I think this ought to be a call to `ros::shutdown()`. 

http://wiki.ros.org/roscpp/Overview/Initialization%20and%20Shutdown#Shutting_Down

Reply:
I would also recommend not calling shutdown in this function, but instead having the function return a bool indicating whether the XML was successfully parsed. Then in the code that calls `parseXML`, handle the `false` case with a call to `std::shutdown()`. 

This allows you to log one final error message before the shutdown, and handle anything else you might need to do before shutdown in only one place instead of 3 places.

Reply:
Done!